homeassistant/components/backup/manager.py

```diff
@@ -240,18 +248,29 @@ def _mkdir_and_generate_backup_contents(
             tar_info.mtime = int(time.time())
             outer_secure_tarfile_tarfile.addfile(tar_info, fileobj=fileobj)
             with outer_secure_tarfile.create_inner_tar(
-                "./homeassistant.tar.gz", gzip=True
+                "./homeassistant.tar.gz",
+                gzip=True,
+                key=_password_to_key(password=password),
             ) as core_tar:
                 atomic_contents_add(
                     tar_file=core_tar,
                     origin_path=Path(self.hass.config.path()),
                     excludes=EXCLUDE_FROM_BACKUP,
                     arcname="data",
                 )
-
         return tar_file_path.stat().st_size
 
 
 def _generate_slug(date: str, name: str) -> str:
     """Generate a backup slug."""
     return hashlib.sha1(f"{date} - {name}".lower().encode()).hexdigest()[:8]
+
+
+def _password_to_key(password: str | None) -> bytes | None:
+    """Generate a AES Key from password."""
+    if password is None:
+        return None
+    key: bytes = password.encode()
+    for _ in range(100):
+        key = hashlib.sha256(key).digest()
```

Comment:
Why 100 rounds here?

Reply:
Because it matches supervisor.... not sure why 100 was chosen here but it needs to be compatible.